PWGJE/Tasks/PhotonIsolationQA.cxx

```diff
@@ -168,17 +174,9 @@ struct PhotonIsolationQA {
   bool track_matching(const auto& cluster, o2::aod::EMCALMatchedTracks const& matched_tracks)
   {
     for (const auto& match : matched_tracks) {
-      double dEta = abs(match.track_as<myGlobTracks>().trackEtaEmcal() - cluster.eta());
-      double dPhi = match.track_as<myGlobTracks>().trackPhiEmcal() - cluster.phi();
-      if (dPhi < 0) {
-        dPhi = 2. * M_PI - dPhi;
-      }
-      double distance = sqrt(pow(dEta, 2) + pow(dPhi, 2));
-      if (distance <= TrackMatchingRadius) {
-        double abs_p = abs(match.track_as<myGlobTracks>().p());
-        if ((cluster.energy() / abs_p) < 1.75) {
-          return true;
-        }
+      double abs_pt = abs(match.track_as<myGlobTracks>().pt());
```

Comment:
I do not suggest to get rid of your track matching requirment in the task itself. What the EMCal track matching in the correction framework ist intended for is to give you the N closest tracks to a given cluster (which is done in a very efficienct way). This means that in your analysis task you do not have to each time loop over all tracks of the event but only those in the vicinity. So in this case, while you are now only looking at the tracks closest to the cluster, you should still make some cut on dEta and dPhi.

Reply:
The emcalCorrection task has a configurable maxMatchingDistance (max matching distance between tracks and cluster). Which is adjustable to different radii. Is that not the same as encoding a dEta and dPhi cut in the PhotonIsolationQA task. Or am i missing something here?

Reply:
the idea is that in the emcal correction framework, this value should be chosen large enough so that it works for every analysis. It is simply used to improve computational efficiency. This is important, because later multiple analyses can run in the same train, but there will be only one correction framework providing the output for all of them. This is why you should choose in the correction framework a large dR and cut in the task on dEta and dPhi of that subset